### ISSUE TITLE
SSL/TLS issues at webhost

### ISSUE THREAD

**Issue by brucew2013** [iATSPayments/com.iatspayments.civicrm#156]
Please confirm: I would like to suggest adding to the README.md file under requirements issues related to this report from my webhost. 

I was getting kicked off IATS due to SSL handshake failing. Due to the use of CentOS 5 anything below TLSv1.2 will not work. Webhost is reimaging my VPS for CentOS 7. There are plenty of webhosts that are not keeping up with upgrades which is why I am suggesting it. For mine, it was an oversight (a mighty one at that ;-) ). 

As follows:
**This looks to be a limitation on your server when connecting to the remote server and we can test this directly using openssl.**

**Shown below is a failed connection to the remote server using TLSv1 (what your wget command specifies) to the remote server's HTTPS port (443):**

```
CT-156611-bash-3.2# openssl s_client -tls1 -connect iatspayments.com:443
CONNECTED(00000003)
31985:error:14094410:SSL routines:SSL3_READ_BYTES:sslv3 alert handshake failure:s3_pkt.c:1092:SSL alert number 40
31985:error:1409E0E5:SSL routines:SSL3_WRITE_BYTES:ssl handshake failure:s3_pkt.c:536:

```
**The same connection fail results from my local workstation. However, when specifying a higher version of TLS such as TLSv1.2, I am able to connect from my workstation as shown below:**

```
$ openssl s_client -quiet -tls1_2 -connect iatspayments.com:443
depth=3 C = SE, O = AddTrust AB, OU = AddTrust External TTP Network, CN = AddTrust External CA Root
verify return:1
depth=2 C = US, ST = New Jersey, L = Jersey City, O = The USERTRUST Network, CN = USERTrust RSA Certification Authority
verify return:1
depth=1 C = US, ST = VA, L = Herndon, O = Network Solutions L.L.C., CN = Network Solutions OV Server CA 2
verify return:1
depth=0 C = US, postalCode = 76131, ST = TX, L = Fort Worth, street = Suite 1800, street = 100 Throckmorton Street, O = First American Payment Systems, OU = Information Technology Group, OU = Secure Link SSL Pro, CN = www.iatspayments.com
verify return:1

```
**Your server is running CentOS 5 which does not support TLS versions higher than TLSv1 and many servers are beginning to reject those connections, such as payment gateways. If you require this support, then it will be necessary to re-image the server to CentOS 7**

**Comment by KarinG** [comment]
The README.md (master) already specifies: "iATS retired TLS1.0 and moved to TLS1.1/1.2 and SHA-256" - we can't possibly add all OS versions that don't support TLS 1.2 - that's for sysadmins to know.

**Comment by adixon** [comment]
It might be worth a general note that you can't and shouldn't be running this payment processor on an old or unmaintained OS. That's advice that is not specific to iATS, though in this particular case it's being enforced indirectly.

So, in this case, it was a valuable feature.

**Comment by adixon** [comment]
Re-opening - should we consider doing a test during the installation process and/or at the top of the console to check this? If it's not hard, seems like a worthwhile addition.

**Comment by lynxlynxlynx** [comment]
Good idea. You can just fork the openssl command from above, but there are several gotchas. Without -quiet, a success means a start of an interactive session. With it, it will just wait and you have to kill it, also destroying the return value. So maybe php has something nicer internally, otherwise `openssl s_client -tls1_2 -connect iatspayments.com:443 <<<"Q"` could do (rc 0 = success).

**Comment by adixon** [comment]
This looks like a simple option: https://wptavern.com/new-wordpress-plugin-tests-for-tls-1-2-compatibility (yeah, i mean, steal someone's code who's already doing something similar, not actually use the plugin).
